Added email_only column to posts_meta table

naz opened the pull request:
closes https://github.com/TryGhost/Team/issues/893
closes https://github.com/TryGhost/Team/issues/894

Reviewers, the only commit for review here is the very first one containing a migration "Added email_only column to posts_meta table " [8729eb2](https://github.com/TryGhost/Ghost/pull/13201/commits/8729eb253f7b266771fa91145cb67ed251873df4). Thanks :pray:

naz committed:
Added email_only column to posts_meta table

closes https://github.com/TryGhost/Team/issues/893

- We need a place to store email-only flag and posts_meta is the best place for it

naz committed:
Added ability to modify email_only property in post resource

https://github.com/TryGhost/Team/issues/893

- The package bump allows to validate the post's `email_only`  property

naz committed:
Added email_only property in Posts Admin API v4

https://github.com/TryGhost/Team/issues/893

- The property is only added to Admin API v4 and is invisible in all Content APIs as well as v2/v3 Posts APIs

allouis commented on core/server/api/canary/utils/serializers/output/utils/mapper.js:
```diff
@@ -52,7 +52,8 @@ const mapPost = (model, frame) => {
     }).each((attr) => {
-        jsonModel[attr] = _.get(jsonModel.posts_meta, attr) || null;
+        const defaultValue = (attr === 'email_only') ? false : null;
+        jsonModel[attr] = _.get(jsonModel.posts_meta, attr) || defaultValue;
```
I'm a bit confused as to the purpose of this, is this to stop a `false` value being turned into a `null`? Or is it because we expect that the value _could_ be `undefined` and we want to add a default?

If it's the first, I wonder if something like this is clearer, in that we check if the value exists, rather than if it's truthy, and use a default if it doesn't
```js
const value = _.get(jsonModel.posts_meta, attr);
jsonModel[attr] = _.isUndefined(value) ? null : value;
```

naz replied:
Nice one @allouis! It's more of the latter one - having an `undefined` for whole posts_meta object and by extension the `email_only` value. The default for the `email_only` is `false` and this is why we want that `if (undefined) assign "false"` logic. 
I see why it's a bit confusing... the ternary operations are not the most readable thing in the world :sweat_smile: 

I'll refactor this into an if/else with a comment. If the question came out now it will definitely confuse us when we come back to this code in the future!

allouis reviewed (commented):
Looks good, just one question about conditional default value stuff

daniellockyer reviewed (approved):
Migration LGTM

naz committed:
Added clarifying comment to the `email_only` mapping

https://github.com/TryGhost/Team/issues/893

- The assignment is not that obvious and might be confusing without wider context, which is why it warrants to have a clarifying comment. This became apparent during code review
